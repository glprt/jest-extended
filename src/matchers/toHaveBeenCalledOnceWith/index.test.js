import matcher from './';

expect.extend(matcher);

describe('.toHaveBeenCalledOnceWith', () => {
  test('should fail when given a mock that has not been called', () => {
    const mock = jest.fn();
    expect(() => expect(mock).toHaveBeenCalledOnceWith()).toThrowErrorMatchingSnapshot();
  });

  test('should fail when given a mock that has been called more than once', () => {
    const mock = jest.fn();
    mock(1);
    mock(2);
    expect(() => expect(mock).toHaveBeenCalledOnceWith()).toThrowErrorMatchingSnapshot();
  });

  test('should pass when given a mock that has been called exactly once with value matching call from mock', () => {
    const mock = jest.fn();
    mock(1);
    expect(mock).toHaveBeenCalledOnceWith(1);
  });

  test('should fail when given a mock that has been called exactly once with one argument but expected multiple', () => {
    const mock = jest.fn();
    mock(1);
    expect(() => expect(mock).toHaveBeenCalledOnceWith(1, 2)).toThrowErrorMatchingSnapshot();
  });

  test('should fail when given a mock that has been called exactly once with multiple arguments but expected one', () => {
    const mock = jest.fn();
    mock(1, 2);
    expect(() => expect(mock).toHaveBeenCalledOnceWith(1)).toThrowErrorMatchingSnapshot();
  });

  test('should pass when given a mock that has been called exactly once with multiple arguments', () => {
    const mock = jest.fn();
    mock(1, 2);
    expect(mock).toHaveBeenCalledOnceWith(1, 2);
  });

  test("should fail when given an expected value that isn't an array", () => {
    const mock = jest.fn();
    mock(1, 2);
    expect(() => expect(mock).toHaveBeenCalledOnceWith(1)).toThrowErrorMatchingSnapshot();
  });
});

describe('.not.toHaveBeenCalledOnceWith', () => {
  test('should pass when given a mock that has not been called', () => {
    const mock = jest.fn();
    expect(mock).not.toHaveBeenCalledOnceWith();
  });

  test('should pass when given a mock that has been called more than once', () => {
    const mock = jest.fn();
    mock(1);
    mock(2);
    expect(mock).not.toHaveBeenCalledOnceWith();
  });

  test('should fail when given a mock that has been called exactly once', () => {
    const mock = jest.fn();
    mock(1);
    expect(() => expect(mock).not.toHaveBeenCalledOnceWith(1)).toThrowErrorMatchingSnapshot();
  });
});
