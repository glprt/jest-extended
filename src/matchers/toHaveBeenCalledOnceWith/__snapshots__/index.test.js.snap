// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`.not.toHaveBeenCalledOnceWith should fail when given a mock that has been called exactly once 1`] = `
"<dim>expect(</><red>received</><dim>).not.toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to not have been called once with:
  <green>[1]</>
Mock called <red>1</> times with:
  <red>[[1]]</>"
`;

exports[`.toHaveBeenCalledOnceWith should fail when given a mock that has been called exactly once with multiple arguments but expected one 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to have been called once with:
  <green>[1]</>
Mock called <red>1</> times with:
  <red>[[1, 2]]</>"
`;

exports[`.toHaveBeenCalledOnceWith should fail when given a mock that has been called exactly once with one argument but expected multiple 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to have been called once with:
  <green>[1, 2]</>
Mock called <red>1</> times with:
  <red>[[1]]</>"
`;

exports[`.toHaveBeenCalledOnceWith should fail when given a mock that has been called more than once 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to have been called once with:
  <green>[]</>
Mock called <red>2</> times with:
  <red>[[1], [2]]</>"
`;

exports[`.toHaveBeenCalledOnceWith should fail when given a mock that has not been called 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to have been called once with:
  <green>[]</>
Mock called <red>0</> times with:
  <red>[]</>"
`;

exports[`.toHaveBeenCalledOnceWith should fail when given an expected value that isn't an array 1`] = `
"<dim>expect(</><red>received</><dim>).toHaveBeenCalledOnce(</><green>expected</><dim>)</>

Expected mock to have been called once with:
  <green>[1]</>
Mock called <red>1</> times with:
  <red>[[1, 2]]</>"
`;
